libr/core/Makefile

```diff
@@ -12,6 +12,7 @@ OBJS=core.o cmd.o file.o cconfig.o visual.o cio.o yank.o libs.o graph.o
 OBJS+=fortune.o hack.o vasm.o patch.o cbin.o log.o rtr.o cmd_api.o
 OBJS+=carg.o canal.o project.o gdiff.o asm.o vmenus.o disasm.o plugin.o
 OBJS+=task.o panels.o pseudo.o vmarks.o anal_tp.o blaze.o cundo.o
+OBJS+=r_core_rabin2.o
```

Comment:
I think the name is a bit too long.

Reply:
its not related to core, in shuold be `r_main_rabin2.o`

Its fine for me to put this inside libr/core, but ideally we may move this into `libr/main/rabin2.c`

Then we can autogenerate all the binr/*/*.c because they will be just an empty main calling the corresponding `r_main_*`. How does it sounds to you? we must include this new libr_main.{a,so} in the libr.{a,so} and link against rcore and all the other libs.

Reply:
in fact we may be able to build them like this:

```
$(CC) -Dr_main_rabin2=main libr/main/rabin2.c -o rabin2
```

Reply:
The main problem of all those mains() are the global variables. we can put a wrapper for the real main() of each tool to fill a struct with those global vars and pass them to the internal main, and the rest of functions that need them. this way we dont break current behaviour and will allow us to run multiple r2 programs in separate threads inside the same process without problems.